CDCgov/prime-reportstream
# Allow normal users to read settings

This PR updates the Settings API to both use server2server auth in addition to Okta auth. This is an addition to #10478 and fixes a bug where a normal org user could not read settings, breaking the daily data page.

Test Steps:
* Run the smoke test

Manual Test:
1. Log in to ReportStream as a normal user (Create a new account in Okta if needed)
2. Verify Daily Data page loads

## Changes
- Updated `SettingsFunctions` `getOne` to support getting an organization better (no longer need to pass in null for orgName, this logic was moved lower down into SettingsFacade)
- Updated getOne and other settings and api key functions to allow for user access (all the ones that are read)

## Linked Issues
- #1946

## Future Work
1. #10810
